src/Middleware/HttpLogging/test/HttpLoggingMiddlewareTests.cs

```diff
@@ -1155,6 +1195,32 @@ public async Task StartAsyncResponseHeadersLogged()
         await middlewareTask;
     }
 
+    [Theory]
+    [MemberData(nameof(BodyData))]
+    public async Task ResponseBodyWithStreamCloseWorks(string expected)
+    {
+        var options = CreateOptionsAccessor();
+        options.CurrentValue.LoggingFields = HttpLoggingFields.ResponseBody;
+        var middleware = CreateMiddleware(
+            async c =>
+            {
+                c.Response.ContentType = "text/plain";
+
+                // (1) The subsequent middleware writes its response
+                await c.Response.WriteAsync(expected);
+
+                // (2) The subsequent middleware closes the response stream after it has completed writing to it
+                c.Response.Body.Close();
+            },
+            options);
+
+        var httpContext = new DefaultHttpContext();
+
+        await middleware.Invoke(httpContext);
+
+        Assert.Contains(TestSink.Writes, w => w.Message.Contains(expected));
+    }
+
```

Comment:
Could we add a test for `options.CombineLogs = true` and both Request and Response?

Reply:
Done